src/Model/Config/NostoConfigService.php

```diff
@@ -147,53 +146,92 @@ public function getConfigWithInheritance(string $salesChannelId = null, string $
+        $queryBuilder = $this->fetchDefaultQueryBuilder($salesChannelId, $languageId);
+        $databaseConfigs = $queryBuilder->executeQuery()->fetchAllNumeric();
 
-        /** @var NostoConfigCollection $collection */
-        $collection = $this->nostoConfigRepository
-            ->search($criteria, Context::createDefaultContext())
-            ->getEntities();
+        if (!count($databaseConfigs)) {
+            return [];
+        }
+
+        $configs = [];
+
+        foreach ($databaseConfigs as [$key, $value]) {
+            if ($value !== null) {
+                $value = json_decode((string) $value, true, 512, \JSON_THROW_ON_ERROR);
+
+                if ($value === false || !isset($value[ConfigJsonField::STORAGE_KEY])) {
+                    $value = null;
+                } else {
+                    $value = $value[ConfigJsonField::STORAGE_KEY];
+                }
+            }
+
+            if ($this->isEmpty($value)) {
+                continue;
+            }
+
+            $configs[$key] = $value;
+        }
 
-        return $this->buildConfig($collection);
+        return $configs;
     }
 
+    /**
+     * @throws Exception
+     */
     public function set(string $key, mixed $value, ?string $salesChannelId = null, ?string $languageId = null): void
     {
         $this->configs = [];
         $key = trim($key);
         $this->validate($key, $salesChannelId, $languageId);
 
         $id = $this->getId($key, $salesChannelId, $languageId);
-        if ($value === null) {
+
+        if ($this->isEmpty($value)) {
             if ($id) {
-                $this->nostoConfigRepository->delete([[
-                    'id' => $id,
-                ]], Context::createDefaultContext());
+                $this->connection->delete(
+                    'nosto_integration_config',
+                    ['id' => Uuid::fromHexToBytes($id)],
+                );
             }
 
             return;
         }
 
-        $this->nostoConfigRepository->upsert(
-            [
+        if ($id) {
+            $this->connection->update(
+                'nosto_integration_config',
                 [
-                    'id' => $id ?? Uuid::randomHex(),
-                    'configurationKey' => $key,
-                    'configurationValue' => $value,
-                    'salesChannelId' => $salesChannelId,
-                    'languageId' => $languageId,
+                    'configuration_value' => Json::encode(['_value' => $value]),
+                    'updated_at' => (new \DateTime())->format(Defaults::STORAGE_DATE_TIME_FORMAT),
                 ],
-            ],
-            Context::createDefaultContext(),
-        );
+                [
+                    'id' => Uuid::fromHexToBytes($id),
+                ]
+            );
+        } else {
+            $this->connection->insert(
+                'nosto_integration_config',
+                [
+                    'id' => $id ?? Uuid::randomBytes(),
```

Comment:
There should be no id at this point
```suggestion
                    'id' => Uuid::randomBytes(),
```